service/common/src/main/java/org/apache/polaris/service/catalog/validation/StorageTypeFileIO.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.catalog.validation;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.polaris.core.storage.PolarisStorageConfigurationInfo;
+
+enum StorageTypeFileIO {
+  S3("org.apache.iceberg.aws.s3.S3FileIO", true),
+
+  GCS("org.apache.iceberg.gcp.gcs.GCSFileIO", true),
+
+  AZURE("org.apache.iceberg.azure.adlsv2.ADLSFileIO", true),
+
+  FILE("org.apache.iceberg.hadoop.HadoopFileIO", false),
```

Comment:
We can't actually guarantee that the implementation found at this location by the classloader is the one you have in mind

Reply:
That is true, but again, I'd suggest improving the "precision" of checks later, after merging this PR.

Reply:
Feature configs are forever -- if we won't be able to make the check precise, we shouldn't add the config. I don't see how the check _can_ be made precise, so I am opposed to adding the config now.

Reply:
Also, `HadoopFileIO` is used for more than just `FILE` -- it's the fallback defined [here](https://github.com/apache/iceberg/blob/97c0e136b8021058897cab7539e3ef89ce5a0341/core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java#L56). In the past we've seen it used for AZURE paths.